x/concentrated-liquidity/incentives_test.go

```diff
@@ -3535,6 +3527,81 @@ func (s *KeeperTestSuite) TestQueryAndClaimAllIncentives() {
 	}
 }
 
+func (s *KeeperTestSuite) TestClaimAllIncentivesForPosition() {
+	testCases := []struct {
+		name             string
+		blockTimeElapsed time.Duration
+		expectedCoins    sdk.Coins
+		expectedForfeit  int
+		expectedErr      bool
+	}{
+		{
+			name:             "Claim with same blocktime",
+			blockTimeElapsed: 0,
+			expectedCoins:    sdk.NewCoins(),
+			expectedForfeit:  0,
+		},
+		{
+			name:             "Claim after 1 minute",
+			blockTimeElapsed: time.Minute,
+			expectedCoins:    sdk.NewCoins(sdk.NewCoin(USDC, sdk.NewInt(59))), //  after 1min = 59.999999999901820104usdc ~ 59usdc becasue 1usdc emitted every second
+			expectedForfeit:  0,
+		},
+		{
+			name:             "Claim after 1 hr",
+			blockTimeElapsed: time.Hour,
+			expectedCoins:    sdk.NewCoins(sdk.NewCoin(USDC, sdk.NewInt(3599))), //  after 1min = 59.999999999901820104usdc ~ 59usdc becasue 1usdc emitted every second
+			expectedForfeit:  0,
+		},
+		{
+			name:             "Claim after 3 hours",
+			blockTimeElapsed: time.Hour * 3,
+			expectedCoins:    sdk.NewCoins(sdk.NewCoin(USDC, sdk.NewInt(9999))), //  after 3hr > 2hr46min = 9999usdc.999999999901820104 ~ 9999usdc
+			expectedForfeit:  0,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			// Init suite for the test.
+			s.SetupTest()
```

Comment:
This is much cleaner, thanks for this. This makes me think though, can we add a test case where we expect some amount to be forfeit? I would imagine this would involve setting one of the uptimes greater than 1ns and then having that test case claim before that time.

Reply:
this makes sense. the only thing is that forfeitedIncentives have been pretty thoroughly tested in `TestQueryAndClaimAllIncentives` https://github.com/osmosis-labs/osmosis/blob/main/x/concentrated-liquidity/incentives_test.go#L2877

this test was just created to check impact of `updatePoolUptimeAccumulatorsToNow` in `claimAllIncentivesForPosition`. Still happy to add it if you think its really important